### PR TITLE
[SM-396] Self-enroll Secrets Manager

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Adds an endpoint for self-enrolling into secrets manager.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Api/Controllers/OrganizationsController.cs [hunk 1]
```diff
@@ -725,4 +726,36 @@ public async Task<OrganizationSsoResponseModel> PostSso(Guid id, [FromBody] Orga
...
+
+        // Turn on Secrets Manager for the user
+        if (model.Enabled)
+        {
+            var orgUser = await _organizationUserRepository.GetByOrganizationAsync(id, userId);
+            orgUser.AccessSecretsManager = true;
+            await _organizationUserRepository.ReplaceAsync(orgUser);
+        }
+
+        return model.Enabled;
```
Comment (shane-melton) [inline src/Api/Controllers/OrganizationsController.cs:759]: question: Do we need to return `model.Enabled` here? We don't appear to use it anywhere client side.

**Review by shane-melton — COMMENTED** [review]
Only had a question about returning the enabled flag when we don't use it anywhere client side. Otherwise, LGTM!

File: src/Api/Controllers/OrganizationsController.cs (hunk 1 above)
Comment (Hinton) [reply src/Api/Controllers/OrganizationsController.cs:759]: Not really. Just seemed better than returning nothing. But we can always add it in the future.

**Review by cd-bitwarden — APPROVED** [review]
Looks good, but do we have a follow up task created to remind us to remove this endpoint yet?

**Pull request merged**